### DISCUSSION TITLE
ImageMagick 7's 'composite' outputs a black & white image, where 6.x didn't for the same data and arguments

### DISCUSSION THREAD

**Discussion by sba923** [ImageMagick/ImageMagick#3053] [Help] [answered]
If I run:

````
C:\Program Files\ImageMagick-6.9.8-Q16\convert.exe -size 4962x7014 canvas:white .\output-v6\canvas.png
C:\Program Files\ImageMagick-6.9.8-Q16\composite.exe -geometry +24+343 S:\mon-timbre-en-ligne\im7bwissue\output-v6\vosTimbres_0_8.png .\output-v6\canvas.png .\output-v6\temp.png
````

I get a color image.

But if I run:

````
C:\Program Files\ImageMagick-7.0.10-Q16-HDRI\convert.exe -size 4962x7014 canvas:white .\output-v7\canvas.png
C:\Program Files\ImageMagick-7.0.10-Q16-HDRI\composite.exe -geometry +24+343 S:\mon-timbre-en-ligne\im7bwissue\output-v7\vosTimbres_0_8.png .\output-v7\canvas.png .\output-v7\temp.png
````

I get a black & white image.

How come?

How does one get the correct / expected result with ImageMagick 7?

FWIW, my repro data can be found [here](https://1drv.ms/u/s!AjBPRo9wXw9s8PwB1eMhwI8JeN9Mzg?e=BrOgEm).

**Comment by urban-warrior** [comment, marked as answer]
See https://imagemagick.org/script/porting.php.  ImageMagick 7 support variable channels.  Your canvas has one gray-scale channel so your output is one channel.  To get the intended results, promote your 1-channel canvas to 3 channels with:

```
composite -geometry +24+343 vosTimbres_0_8.png canvas.png -type truecolor temp.png
```

**Reply by sba923** [reply]
Thanks, this works like a charm!

And thanks too for the "v7 porting guide" link. I'll keep that handy!
